torch/distributed/distributed_c10d.py

```diff
@@ -172,6 +173,22 @@ class GroupMember(object):
 # Process group count for default naming
 _group_count = 0
 
+STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
+
+def _store_based_barrier(rank, store, timeout):
+    global _group_count
+    store_key = "{}:{}".format(STORE_BASED_BARRIER_PREFIX, _group_count)
+    store.add(store_key, 1)
+
+    # Now wait for all workers to check in with the store.
+    world_size = get_world_size()
+    worker_count = int(store.get(store_key))
+    start = time.time()
+    while worker_count != world_size:
```

Comment:
this means `_store_based_barrier` can only be called once for each pg? Shall we throw proper error if somehow `worker_count` becomes larger than `world_size`?

Reply:
`_store_based_barrier` is a private method for now and only intended for use with init_process_group/new_group which is only done once per process group.